src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -313,6 +332,11 @@ bool ITaskExecutionHost.InitializeForBatch(TaskLoggingContext loggingContext, It
                 return false;
             }
 
+            if (TaskInstance.GetType().Assembly.Location != _taskFactoryWrapper.TaskFactoryLoadedType.Path)
```

Comment:
nit: Is the string equality comparison OK here? Should this be something like `FileUtilities.ComparePathsNoThrow` instead?

Reply:
Value set to `TaskFactoryLoadedType.Path` is from `Assembly.Location`. During OrchadCore build, all locations were equal (except in-memory assemblies).

Reply:
Gotcha. So even if the project has `UsingTask`'s with different casing this would probably work fine because the path is normalized when we load the task assembly. I have no objections then!

Reply:
Actually - is it possible to cover this with a test?

Reply:
I will take a deeper look. Finding the right test to extend is not always an easy task :)

Reply:
We agreed with @ladipro (offline) that we will skip the test due the complexity, but the message was changed.